File: backend/ws-service/src/main/java/com/example/wsservice/chat/service/ChatService.java
package com.example.wsservice.chat.service;

import com.example.wsservice.chat.client.ChatFeignClient;
import com.example.wsservice.chat.dto.ChatDto;
import com.example.wsservice.chat.dto.ReactionRequestDto;
import com.example.wsservice.chat.dto.ReactionResultDto;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class ChatService {

    private final ChatFeignClient chatFeignClient;

    public Long saveChat(ChatDto chatDto) {
        return chatFeignClient.saveChat(chatDto);
    }

    public ChatDto getChat(String chatId) {
        return chatFeignClient.getChat(chatId);
    }

    /** 반응 토글. chat-service 가 거부하면(내 메시지·모르는 이모지) FeignException 이 난다. */
    public ReactionResultDto react(String roomId, String chatId, String userId, String emoji) {
        return chatFeignClient.react(roomId, chatId, userId, new ReactionRequestDto(emoji));
    }
}

File: backend/ws-service/src/main/kotlin/com/example/wsservice/chat/service/ChatService.kt
package com.example.wsservice.chat.service

import com.example.wsservice.chat.client.ChatFeignClient
import com.example.wsservice.chat.dto.ChatDto
import com.example.wsservice.chat.dto.ReactionRequestDto
import com.example.wsservice.chat.dto.ReactionResultDto
import org.springframework.stereotype.Service

@Service
class ChatService(private val chatFeignClient: ChatFeignClient) {

    fun saveChat(chatDto: ChatDto): Long = chatFeignClient.saveChat(chatDto)

    fun getChat(chatId: String): ChatDto = chatFeignClient.getChat(chatId)

    /** 반응 토글. chat-service 가 거부하면(내 메시지·모르는 이모지) FeignException 이 난다. */
    fun react(roomId: String, chatId: String, userId: String, emoji: String): ReactionResultDto =
        chatFeignClient.react(roomId, chatId, userId, ReactionRequestDto(emoji))
}
